site/source/tutorials/1010_containers_and_injecting_state/index.md

```diff
@@ -0,0 +1,175 @@
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' line:31-32 %}
+
+The first statement creates a `registry` where the application context can be registered. The second statement does two things. First, it uses the `Injector` factory function to wrap the `applicationContext` with the `BaseInjector`, a lightweight component that allows any object that extends `Evented` to be injected via a `Container`, which we will discuss in the next section. This statement also registers the `Injector` with the application's central registry. The registry provides a way to register a widget via a label, making it accessible to other parts of the application. You can learn more in the [registry tutorial](../comingsoon.html).
+
+{% include_codefile 'demo/finished/biz-e-corp/src/main.ts' line:36 %}
+
+We need to pass the `registry` to the `projector` via the `setProperties` method to ensure that it is available for all widget and container instances.
+
+Now that the `Injector` is defined and registered and the `registry` has been set in the `projector`, it is time to create the components that will use it. In the next section, we will create a non-visual widget called a `Container` that will allow state to be injected into the `WorkerForm` and `WorkerContainer` widgets.
+
+{% section %}
+
+## Creating state containers
+
+{% task 'Create `Containers` that will allow state to be injected into widgets' %}
+
+On their own, `Injectors` are not able to help us very much because widgets expect state to be passed to them via properties. Therefore an `injector` must be connected to interested widgets in order for their state to be mapped to `properties` that widgets can consume, this is done using a `Container`. `Containers` are designed to coordinate the injection - they connect injectors to widgets and return `properties` from the `injectors` state which gets passed to the connected widgets.
+
+Normally, a separate `Container` is created for each widget that needs to have `properties` injected. In the demo application, we have two widgets that rely on application state - `WorkerContainer` and `WorkerForm`. Let's start with the `WorkerContainer`. `Containers` are given the same name as the widget that they contain with the suffix `Container` appended. To keep things organized, they are also stored in a different sub-package - `containers`.
+
+{% instruction 'Add the following imports to the `WorkerContainerContainer` in the `containers` sub-package' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/containers/WorkerContainerContainer.ts' lines:1-4 %}
+
+The first `import` gives the module access to the `Container` factory function which will be used to construct the container. The other `import` allow the module to use the `ApplicationContext` to extract state, use the `WorkerContainerProperties` to receive properties from its parent, and the `WorkerContainer` class that will be wrapped by the container.
```

Comment:
"and the `WorkerContainer` class that will be wrapped by the container." -> "and wrap the `WorkerContainer` class with the `container`."